shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -396,17 +396,24 @@ - (void)setupChannels {
 
 - (void)maybeSetupPlatformViewChannels {
   if (_shell && self.shell.IsSetup()) {
+    fml::WeakPtr<FlutterEngine> weakSelf = [self getWeakPtr];
     [_platformChannel.get() setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
-      [_platformPlugin.get() handleMethodCall:call result:result];
+      if (weakSelf) {
```

Comment:
@zhongwuzw Can you change this to:
```objc
FlutterPlatformPlugin* platformPlugin = weakSelf.get().platformPlugin;
[_platformChannel.get() setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
  [platformPlugin handleMethodCall:call result:result];
}];
```

Same for the text plugin.  That would have better performance and break the retain cycle.

Reply:
@gaaclarke Done.